### DISCUSSION TITLE
Caching issues with newer luajit?

### DISCUSSION THREAD

**Discussion by swarn** [neovim/neovim#25265] [Q&A] [answered]
Commit e99a3fd bumped luajit to commit 5a18d4582. I have a custom colorscheme `mycolors` defined via lush.nvim and loaded via lazy.nvim. When I load it with `colorscheme mycolors`, I get the error below.

The tricky part is, if I `touch mycolors.lua` to update the file modified time, then the error won't occur the _first_ time I use `colorscheme mycolors`. It will then occur every time until I update the file modified time again. So, I assume this is something to do with caching compiled lua files.

Obviously there are quite a few moving parts in this description, and both lush and lazy are doing some tricky stuff. Before putting in the effort to create a minimal example for an actual issue, I thought I'd check to see if this was a known problem.

```
Error detected while processing /Users/seth/src/edit/mycolors/colors/mycolors.lua:
E5113: Error while calling lua chunk: ...eth/.local/share/nvim/lazy/lush.nvim/lua/lush/parser.lu
a:109: bad argument #1 to 'unpack' (table expected, got nil)
stack traceback:
        [C]: in function 'unpack'
        ...eth/.local/share/nvim/lazy/lush.nvim/lua/lush/parser.lua:109: in function 'validator'

        ...eth/.local/share/nvim/lazy/lush.nvim/lua/lush/parser.lua:194: in function 'enforce'
        ...eth/.local/share/nvim/lazy/lush.nvim/lua/lush/parser.lua:262: in function 'create_gro
up'
        ...eth/.local/share/nvim/lazy/lush.nvim/lua/lush/parser.lua:432: in function 'Comment'
        /Users/seth/src/edit/mycolors/lua/lush_theme/mycolors.lua:239: in function 'lush_spec_fn
'
        ...eth/.local/share/nvim/lazy/lush.nvim/lua/lush/parser.lua:537: in function 'parse'
        /Users/seth/.local/share/nvim/lazy/lush.nvim/lua/lush.lua:76: in function 'lush'
        /Users/seth/src/edit/mycolors/lua/lush_theme/mycolors.lua:93: in main chunk
        [C]: in function 'require'
        /Users/seth/src/edit/mycolors/colors/mycolors.lua:20: in main chunk
```

**Comment by clason** [comment]
Have you bisected the issue (bumping back to a previous version while keeping the rest unchanged)? Of course caching is not expected to be stable between LuaJIT commits, but if you wipe the cache, there should be no error from then on.

In general, this sounds unrelated to LuaJIT, at least in the absence of further information. In any case, a minimal reproducer for a Neovim issue should not involve lazy (otherwise it's *prima facie* a lazy issue).

**Comment by swarn** [comment]
I bisected the nvim commits: e99a3fd has the issue described, but the previous commit 59d9f2413 does not have the issue, with no other changes in config.

Certainly a minimal example will omit lazy.nvim, when I get the chance.

**Reply by clason** [reply]
I really meant the LuaJIT commits.

**Reply by swarn** [reply]
Is there an easy way to do this? I can `git bisect` in my neovim repo to easily try neovim commits, but the only way I know to try different luajit versions with the same nvim is to manually edit the LUAJIT_URL in deps.txt.

**Reply by clason** [reply]
Edit `cmake.deps/deps.txt` (you can point to a local checked out repo, then you don't need to worry about the hash; just `make distclean` after each `git bisect` step in the LuaJIT repo).

**Reply by swarn** [reply]
Well now, this is fascinating. The commit that breaks it is https://github.com/LuaJIT/LuaJIT/commit/435d8c630135d4f6a54f2ecf7be2d7e805652f80

That's ARM64-specific (and I'm on mac). Current neovim HEAD works just fine pointing to the previous luajit commit in `deps.txt`.

**Reply by clason** [reply, marked as answer]
So it's a LuaJIT issue. If you are brave (and ready to do some homework), you can open an issue there.

**Reply by swarn** [reply]
Indeed it is, and that repo will be the proper place for an issue.

Thanks for helping me narrow it down!

Re: Matrix, am I the only one that dislikes it? This conversation with its half-hour gaps would have had 100 other interleaved messages about setting up auto formatting :). Unless there's a way to create conversation threads...

**Reply by clason** [reply]
Threads are even more hateful, but there are lower-volume channels for such technical discussions.

**Comment by clason** [comment]
(Also, such discussions are better had on Matrix.)

**Comment by swarn** [comment]
Fixed in LuaJIT/LuaJIT@91592899275cbb540ca67bbf95b41a2200e4fdbd

**Comment by clason** [comment]
Already in Neovim master.
